src/FBAuthenticationToken.js

```diff
@@ -0,0 +1,64 @@
+/**
+ * @flow
+ * @format
+ */
+
+ 'use strict';
+
+const { Platform, NativeModules } = require('react-native');
+ const AuthenticationToken = NativeModules.FBAuthenticationToken;
+
+ type AuthenticationTokenMap = {
+   authenticationToken: string,
+   nonce: string,
+   graphDomain: string,
+ };
+ 
+ /**
+  * Represents an immutable access token for using Facebook services.
+  */
+ class FBAuthenticationToken {
+   
+   /**
+     The raw token string from the authentication response
+    */
+   authenticationToken: string;
+
+   /**
+     The nonce from the decoded authentication response
+    */
+   nonce: string;
+
+   /**
+    The graph domain where the user is authenticated.
+   */
+   graphDomain: string;
+ 
+   constructor(tokenMap: AuthenticationTokenMap) {
+     this.authenticationToken = tokenMap.authenticationToken;
+     this.nonce = tokenMap.nonce;
+     this.graphDomain = tokenMap.graphDomain;
+     Object.freeze(this);
+   }
+ 
+   /**
+    * Getter for the authentication token
+    */
+   static getAuthenticationToken(): Promise<?FBAuthenticationToken> {
+     if (Platform.OS === 'android') {
+       return Promise.resolve(null);
+      }
```

Comment:
I understand AuthenticationToken only valid in iOS, but I think we can give more information to the developers about this platform-specific function via name or warnings. Something like `-IOS` suffix perhaps? e.g. `getAuthenticationTokenIOS`